source/arm9/sound.c

```diff
@@ -122,6 +135,8 @@ MicCallback micCallback = 0;
 void micBufferHandler(int bytes, void* user_data){
 	FifoMessage msg;
 
+	// This is inside an IRQ, no locks required.
```

Comment:
That's not what I meant! fifoWaitValueAsync32 will call cothread_yield, and I'm not sure if our yielding routine handles cothread_yield being called from inside an IRQ. My proposal was to keep this one blocking for the time being.